charts/renku/templates/_keycloak-clients-users.tpl

```diff
@@ -0,0 +1,47 @@
+{{/* vim: set filetype=mustache: */}}
+{{/*
+Define clients and users for Keycloak
+*/}}
+{{- define "renku.keycloak.clients" -}}
+[
+  {
+    "clientId": "gateway",
+    "baseUrl": "{{ template "http" . }}://{{ .Values.global.renku.domain }}/api",
+    "secret": "{{ required "Fill in .Values.global.gateway.clientSecret with `uuidgen -r`" .Values.global.gateway.clientSecret }}",
+    "redirectUris": [
+        "{{ template "http" . }}://{{ .Values.global.renku.domain }}/*"
+    ],
+    "webOrigins": [
+        "{{ template "http" . }}://{{ .Values.global.renku.domain }}/*"
+    ]
+  }
+
+  {{- if .Values.gitlab.enabled -}}
+  ,{
+    "clientId": "gitlab",
+    "baseUrl": "{{ template "http" . }}://{{ .Values.global.renku.domain }}/gitlab",
+    "secret": "{{ required "Fill in .Values.global.gitlab.clientSecret with `uuidgen -r`" .Values.global.gitlab.clientSecret }}",
+    "redirectUris": [
+      "{{ template "http" . }}://{{ .Values.global.renku.domain }}/gitlab/users/auth/oauth2_generic/callback"
+    ],
+    "webOrigins": []
+  }
+  {{- end -}}
+]
+{{- end -}}
+
+{{- define "renku.keycloak.users" -}}
+[
+  {{- if .Values.keycloak.createDemoUser -}}
+  {
+    "username": "demo",
+    "password": "demo",
```

Comment:
maybe generate a random password and provide the `kubectl` command needed to retrieve it?

Reply:
Since the demo user is not created by default I thought that an "all-open" demo/demo user would be preferable, but if there are strong concerns about this, I'm happy to follow your suggestion.

Reply:
it makes debugging more annoying, but the user is not removed even it `createDemoUser=false` in subsequent deployments so it's likely to just stay there until someone actively disables/removes it.

Reply:
Ok, I did as you suggested. Note that for the moment the correct password for the demo user can only be retrieved using `kubectl` after the initial install. On upgrade the corresponding secret will be overwritten, but not the password of the demo user. This will be fixed when resolving https://github.com/SwissDataScienceCenter/renku/pull/647 which is already in progress.